src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1683,10 +1683,31 @@
   <data name="DetectPreviewFeaturesTitle" xml:space="preserve">
     <value>This API requires opting into preview features</value>
   </data>
-  <data name="ImplementsEmptyPreviewInterfaceMessage" xml:space="preserve">
-    <value>'{0}' implements a preview interface. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
-  </data>
   <data name="StaticAndAbstractRequiresPreviewFeatures" xml:space="preserve">
     <value>Using both 'static' and 'abstract' modifiers requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
   </data>
+  <data name="DerivesFromPreviewClassMessage" xml:space="preserve">
+    <value>'{0}' derives from preview class '{1}' and therefore needs to opt into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
+  </data>
+  <data name="ImplementsPreviewInterfaceMessage" xml:space="preserve">
+    <value>'{0}' implements the preview interface '{1}' and therefore needs to opt into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
+  </data>
+  <data name="ImplementsPreviewMethodMessage" xml:space="preserve">
+    <value>'{0}' implements the preview method '{1}' and therefore needs to opt into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
+  </data>
+  <data name="MethodReturnsPreviewTypeMessage" xml:space="preserve">
+    <value>'{0}' returns the preview type '{1}' and therefore needs to opt into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
+  </data>
+  <data name="MethodUsesPreviewTypeAsParamaterMessage" xml:space="preserve">
+    <value>`{0}` takes in a preview parameter of type `{1}` and needs to opt into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.</value>
```

Comment:
Change backtick to single quote for consistency

Reply:
Can you hold off on reviewing for a few minutes? I'm just going through the diff on GH myself. I'll ping you once I'm done!!

Reply:
Ok done, can be reviewed now @Youssef1313